src/openssl/ssl/context.cr

```diff
@@ -77,6 +77,24 @@ abstract class OpenSSL::SSL::Context
     def self.insecure(method : LibSSL::SSLMethod = LibSSL.sslv23_method)
       super(method)
     end
+
+    # Wraps the original certificate verification to also validate the
+    # hostname against the certificate configured Subject Alternate
+    # Names or Common Name.
+    #
+    # Required for OpenSSL <= 1.0.1 only. Newer versions should use
+    # `LibSSL.x509_verify_param_set1_ip_asc` and
+    # `LibSSL.x509_verify_param_set1_host` instead.
```

Comment:
Might be worth to mention that `OpenSSL::SSL::Socket` does use either automatically.